java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -6701,26 +6701,27 @@ private ColumnView getColumnViewWithNonEmptyNulls() {
     List<Integer> list1 = Arrays.asList(4, 5, null);
     List<Integer> list2 = Arrays.asList(7, 8, 9);
     List<Integer> list3 = null;
-    ColumnVector input =  makeListsColumn(DType.INT32, list0, list1, list2, list3);
-
-    // Modify the validity buffer
-    BaseDeviceMemoryBuffer dmb = input.getDeviceBufferFor(BufferType.VALIDITY);
-    HostMemoryBuffer newValidity = HostMemoryBuffer.allocate(64);
-    newValidity.copyFromDeviceBuffer(dmb);
-    BitVectorHelper.setNullAt(newValidity, 1);
-    dmb.copyFromHostBuffer(newValidity);
-
-    HostColumnVector hostColumnVector = input.copyToHost();
-    assert(hostColumnVector.isNull(1));
-    assert(hostColumnVector.isNull(3));
-
-    ColumnVector expectedOffsetsBeforePurge = ColumnVector.fromInts(0, 3, 6, 9, 9);
-    ColumnView offsetsCvBeforePurge = input.getListOffsetsView();
-    assertColumnsAreEqual(expectedOffsetsBeforePurge, offsetsCvBeforePurge);
-    ColumnView colWithNonEmptyNulls = new ColumnView(input.type, input.rows, Optional.of(2L), dmb,
-        input.getDeviceBufferFor(BufferType.OFFSET), input.getChildColumnViews());
-    assertEquals(2, colWithNonEmptyNulls.nullCount);
-    return colWithNonEmptyNulls;
+    try (ColumnVector input = makeListsColumn(DType.INT32, list0, list1, list2, list3)) {
+      // Modify the validity buffer
+      BaseDeviceMemoryBuffer dmb = input.getDeviceBufferFor(BufferType.VALIDITY);
+      try (HostMemoryBuffer newValidity = HostMemoryBuffer.allocate(64)) {
+        newValidity.copyFromDeviceBuffer(dmb);
+        BitVectorHelper.setNullAt(newValidity, 1);
+        dmb.copyFromHostBuffer(newValidity);
+      }
+      try (HostColumnVector hostColumnVector = input.copyToHost()) {
+        assert (hostColumnVector.isNull(1));
+        assert (hostColumnVector.isNull(3));
+      }
+      try (ColumnVector expectedOffsetsBeforePurge = ColumnVector.fromInts(0, 3, 6, 9, 9)) {
+        ColumnView offsetsCvBeforePurge = input.getListOffsetsView();
+        assertColumnsAreEqual(expectedOffsetsBeforePurge, offsetsCvBeforePurge);
+      }
+      ColumnView colWithNonEmptyNulls = new ColumnView(input.type, input.rows, Optional.of(2L), dmb,
+          input.getDeviceBufferFor(BufferType.OFFSET), input.getChildColumnViews());
+      assertEquals(2, colWithNonEmptyNulls.nullCount);
+      return colWithNonEmptyNulls;
```

Comment:
I don't think this returned vector should be in this try block.

Reply:
That's not the problem, the problem is that the `input` column will be closed while the view won't. Unfortunately my build env is completely ruined and I will have to spend time to fix it before I can proceed. Thanks for your quick reviews